File: my-folder/0238-product-of-array-except-self/solution.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        array_length = len(nums)
        result = [1] * array_length

        left_product = 1
        right_product = 1

        for i in range(array_length):
            result[i] = left_product
            left_product *= nums[i]

        for i in range(array_length - 1, -1, -1):
            result[i] *= right_product
            right_product *= nums[i]
        
        return result

File: my-folder/0238-product-of-array-except-self/solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    
    prefix = 1;
    postfix = 1;
    result = new Array(nums.length).fill(1);

    for (let i=0; i < nums.length; i++) {
        result[i] = prefix
        prefix *= nums[i]

    }    
    for (let i=nums.length -1; i >= 0; i--){
        result[i] *= postfix;
        postfix *= nums[i] 
    }
    return result
};
